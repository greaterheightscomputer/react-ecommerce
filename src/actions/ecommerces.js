import uuid from 'uuid';

//Add_Ecommerce
export const addEcommerce = ({ description='', category='', item='', amount=0, image='', imageUrl='', stock=0, createdAt=0 } = {}) => ({
    type: 'ADD_ECOMMERCE',
    ecommerce: {
        id: uuid(),        
        description,
        category,
        item,
        amount,
        image,
        imageUrl,
        stock,
        createdAt
    }
});
// REMOVE_Ecommerce    
export const removeEcommerce = ( { id } = {}) => ({
    type: 'REMOVE_ECOMMERCE',
    id
});
// EDIT_Ecommerce  
export const editEcommerce = (id, updates) => ({
    type: 'EDIT_ECOMMERCE',
    id,
    updates
});
