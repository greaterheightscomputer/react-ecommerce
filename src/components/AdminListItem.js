import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const AdminListItem = ({ id, description, amount, imageUrl, stock}) => (          
    <div>
        <Link to={`/edit/${id}`}>
            <img  src={imageUrl} height="200" width="150" /> }                       
            <h3>{description}</h3>
            <div> 
                {'â‚¦'+numeral(amount / 100).format('0,0.00')}             
                {stock === "0" ? (<p>Out of stock</p>) : (<p>{`${stock} piece(s) available`}</p>)}                            
            </div>
            
            
               
        </Link>   
    </div>
);

export default AdminListItem;