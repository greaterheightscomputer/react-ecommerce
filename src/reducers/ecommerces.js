const ecommerceReducerDefaultState = [];
const ecommerceReducer = (state = ecommerceReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_ECOMMERCE':            
            return [...state, action.ecommerce];   
        case 'REMOVE_ECOMMERCE':            
            return state.filter(({ id }) => id !== action.id); 
        case 'EDIT_ECOMMERCE':
            return state.map((ecommerce) => {
                if (ecommerce.id === action.id){
                    return {...ecommerce, ...action.updates};
                }else {
                    return ecommerce;
                }
            });              
        default: return state;        
    }
};

export default ecommerceReducer;