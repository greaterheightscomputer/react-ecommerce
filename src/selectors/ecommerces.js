import moment from "moment";

//Get visible ecommerce -> this function will match the ecommerce array object with the filters object
export default (ecommerces, { text, sortBy, itemType,  categoryType, startDate, endDate }) => {
    return ecommerces.filter((ecommerce) => {  //filter() allow us to return a subset of all of the array expenses.
        // const startDateMatch = typeof startDate !== 'number' || ecommerce.createdAt >= startDate;
        // const endDateMatch = typeof endDate !== 'number' || ecommerce.createdAt <= endDate;
        const createdAtMoment = moment(ecommerce.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = ecommerce.description.toLowerCase().includes(text.toLowerCase());
        const categoryMatch = ecommerce.category.includes(categoryType);
        const itemMatch = ecommerce.item.includes(itemType);
        
        return startDateMatch && endDateMatch && textMatch && categoryMatch && itemMatch;
    }).sort((a, b) =>{    //compare function. sort() get called on a return subset array. if a < b return -1 else if a > b return 1  else a === b return 0
          if(sortBy === 'date') {
              return a.createdAt < b.createdAt ? 1 : -1;   // the truthy value is 1 which means b is more recent than a 
        }                                                  
        if(sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
      }
    });
};

