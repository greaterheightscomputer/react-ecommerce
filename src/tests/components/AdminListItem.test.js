import React from 'react';
import { shallow } from 'enzyme';
import ecommerces from '../fixtures/ecommerces';
import AdminListItem from '../../components/AdminListItem';

test('should render EcommerceListItem correctly', () => {
    const wrapper = shallow(<AdminListItem {...ecommerces[2]} />);
    expect(wrapper).toMatchSnapshot();
});