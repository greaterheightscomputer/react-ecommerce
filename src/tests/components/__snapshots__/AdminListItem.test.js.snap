// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EcommerceListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3"
  >
    <img
      height="200"
      src=""
      width="150"
    />
    <h3>
      boy shirt
    </h3>
    <div>
      â‚¦3.50
      <p>
        3200 piece(s) available
      </p>
    </div>
  </Link>
</div>
`;
